File: python/exercise1/palindrome.py
def is_palindrome(string):
    while len(string) > 1:
        a = string[0]
        b = string[-1]
        string = string[1:-1]

        if a != b:
            return False

    if len(string) == 0:
        raise Exception("FIX ME PLEASE")

    return True

File: cpp/exercise1/palindrome.cpp
#include <string>
#include <exception>

bool is_palindrome(std::string str)
{

    int left = 0;
    int right = str.length() - 1;

    while (true)
    {
        if (left >= right)
        {
            break;
        }

        char left_char = str[left];
        char right_char = str[right];

        if (left_char != right_char)
        {
            return false;
        }

        ++left;
        --right;
    }

    return true;
}
